### ISSUE TITLE
New Czech TV web-site does not work

### ISSUE THREAD

**Issue by MyloshB** [ytdl-org/youtube-dl#30696]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2021.12.17**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--no-check-certificate', '--verbose', '--no-overwrites', '--restrict-filenames', '--output', 'C:\\Temp\\%(title)s_%(autonumber)s.%(ext)s', '--batch-file', 'C:\\Program Files (x86)\\YouTube-DL\\youtube-dl.lst']
[debug] Batch file urls: ['www.ceskatelevize.cz/ivysilani/embed/iFramePlayer.php?hash=18f5f760865c2ea87d0952ff185d2983a060cc58']
[debug] Encodings: locale cp1250, fs mbcs, out cp852, pref cp1250
[debug] youtube-dl version 2021.12.17
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.19041
[debug] exe versions: ffmpeg 4.3.2-2021-02-20-full_build-www.gyan.dev, ffprobe 4.3.2-2021-02-20-full_build-www.gyan.dev
[debug] Proxy map: {}
WARNING: The url doesn't specify the protocol, trying with http
[CeskaTelevize] iFramePlayer.php: Downloading webpage
ERROR: Unable to extract type; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\YoutubeDL.py", line 815, in wrapper
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\YoutubeDL.py", line 836, in __extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\extractor\common.py", line 534, in extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\extractor\ceskatelevize.py", line 94, in _real_extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\extractor\common.py", line 1021, in _html_search_regex
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\extractor\common.py", line 1012, in _search_regex
youtube_dl.utils.RegexNotFoundError: Unable to extract type; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

## Description

New formats of Czech TV video-files does not work...
Example 1: 
https://www.ceskatelevize.cz/porady/13875183193-zappa/22138775999/
Example 2:
https://www.ceskatelevize.cz/ivysilani/embed/iFramePlayer.php?hash=18f5f760865c2ea87d0952ff185d2983a060cc58&origin=iVysilani&autoStart=true&width=100%25&IDEC=22138775999&start=0

**Comment by Vangelis66** [comment]
I tried "**Example 1**" sample URI and got an error similar to the one in #30406, so, 
most probably, this is a **duplicate** of that one... 😉 

FWIW, latest daily/nightly `yt-dlp` does seem able to detect & extract available formats, 
```
yt-dlp -F "https://www.ceskatelevize.cz/porady/13875183193-zappa/22138775999/" => 
[CeskaTelevize] 22138775999: Downloading webpage
[CeskaTelevize] 22138775999: Downloading webpage
[CeskaTelevize] 22138775999: Downloading webpage
[CeskaTelevize] 22138775999: Downloading JSON metadata
[CeskaTelevize] 22138775999: Downloading JSON metadata
[CeskaTelevize] 22138775999: Downloading MPD manifest
[CeskaTelevize] 22138775999: Downloading JSON metadata
[CeskaTelevize] 22138775999: Downloading JSON metadata
[CeskaTelevize] 22138775999: Downloading m3u8 information
[download] Downloading playlist: Zappa - iVys?l?n? | ?esk? televize
[CeskaTelevize] playlist Zappa - iVys?l?n? | ?esk? televize: Collected 1 videos; downloading 1 of them
[download] Downloading video 1 of 1
[info] Available formats for 61924494878000343:
ID                  EXT RESOLUTION |   FILESIZE   TBR PROTO  | VCODEC        VBR ACODEC     ABR      ASR             MORE INFO
-------------------------------------------------------------------------------------------------------------------------------------
dash-main-1002-1504 m4a audio only | ~ 85.97MiB   96k dash   | audio only mp4a.40.2         96k    44100Hz [ces] DASH audio, m4a_dash
hls-main-628        mp4 unknown    | ~562.38MiB  628k m3u8_n | unknown      628k unknown     0k
hls-main-1160       mp4 unknown    | ~  1.01GiB 1160k m3u8_n | unknown     1160k unknown     0k
hls-main-2176       mp4 unknown    | ~  1.90GiB 2176k m3u8_n | unknown     2176k unknown     0k
dash-main-1001-1502 mp4 512x288    | ~476.41MiB  532k dash   | avc1.4d4015  532k video only                      DASH video, mp4_dash
dash-main-1001-1503 mp4 720x404    | ~952.82MiB 1064k dash   | avc1.4d401e 1064k video only                      DASH video, mp4_dash
dash-main-1001-1504 mp4 1024x576   | ~  1.82GiB 2080k dash   | avc1.4d401f 2080k video only                      DASH video, mp4_dash
[download] Finished downloading playlist: Zappa - iVys?l?n? | ?esk? televize
```
... but unless you have a whitelisted **Czech IP address**, any one of available formats will fetch an 11sec worth **short clip** informing you of the **geo-block**...

**Comment by dirkf** [comment]
I believe it is the same problem and will be fixed by the enhanced back-ported yt-dlp extractor that's in the works:
```
$ python -m youtube_dl -F -v 'https://www.ceskatelevize.cz/porady/13875183193-zappa/22138775999/'
[debug] System config: [u'--prefer-ffmpeg']
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-F', u'-v', u'https://www.ceskatelevize.cz/porady/13875183193-zappa/22138775999/']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Git HEAD: 6508688e8
[debug] Python version 2.7.17 (CPython) - Linux-4.4.0-210-generic-i686-with-Ubuntu-16.04-xenial
[debug] exe versions: avconv 4.3, avprobe 4.3, ffmpeg 4.3, ffprobe 4.3
[debug] Proxy map: {}
[CeskaTelevize] 22138775999: Downloading webpage
[CeskaTelevize] 22138775999: Getting IFRAME hash
[CeskaTelevize] 22138775999: Downloading player
[CeskaTelevize] 22138775999: Downloading JSON metadata
[CeskaTelevize] 22138775999: Downloading JSON metadata
[CeskaTelevize] 22138775999: Downloading MPD manifest
[CeskaTelevize] 22138775999: Downloading JSON metadata
[CeskaTelevize] 22138775999: Downloading JSON metadata
[CeskaTelevize] 22138775999: Downloading m3u8 information
[info] Available formats for 61924494878000343:
format code          extension  resolution note
dash-main-1002-1504  m4a        audio only [ces] DASH audio   96k , m4a_dash container, mp4a.40.2 (44100Hz)
dash-main-1001-1502  mp4        512x288    DASH video  532k , mp4_dash container, avc1.4d4015, video only
dash-main-1001-1503  mp4        720x404    DASH video 1064k , mp4_dash container, avc1.4d401e, video only
dash-main-1001-1504  mp4        1024x576   DASH video 2080k , mp4_dash container, avc1.4d401f, video only
hls-main-628         mp4        unknown     628k 
hls-main-1160        mp4        unknown    1160k 
hls-main-2176        mp4        unknown    2176k  (best)
$
```
